source/server/listener_manager_impl.cc

```diff
@@ -223,21 +223,29 @@ ListenerImpl::ListenerImpl(const envoy::api::v2::Listener& config, const std::st
     addListenSocketOptions(
         Network::SocketOptionFactory::buildLiteralOptions(config.socket_options()));
   }
+  bool is_quic{false};
+
   if (socket_type_ == Network::Address::SocketType::Datagram) {
     // Needed for recvmsg to return destination address in IP header.
     addListenSocketOptions(Network::SocketOptionFactory::buildIpPacketInfoOptions());
     // Needed to return receive buffer overflown indicator.
     addListenSocketOptions(Network::SocketOptionFactory::buildRxQueueOverFlowOptions());
     std::string listener_name =
         config.has_udp_listener_config() ? config.udp_listener_config().udp_listener_name() : "";
+    is_quic = listener_name == UdpListenerNames::get().Quic;
```

Comment:
nit: is_quic = config.has_udp_listener_config() && config.udp_listener_config().udp_listener_name() == UdpListenerNames::get().Quic;

Also you can create const ref of udp_listener_config here and declare listener_name const

const auto & udp_config = config.has_udp_listener_config() ? config.udp_listener_config() : envoy::api::v2::listener::UdpListenerConfig();
const auto  listener_name = udp_config.udp_listener_name().empty() ? udp_config.udp_listener_name() : "";

The idea is to figure out the final listener_name at one place